Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -139,11 +139,14 @@ + (WKWebViewConfiguration *)configuration
     if (!configuration) {
         configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
         [configuration setProcessPool:adoptNS([[WKProcessPool alloc] init]).get()];
-        [configuration setWebsiteDataStore:[WKWebsiteDataStore nonPersistentDataStore]];
+        [configuration setWebsiteDataStore:[WKWebsiteDataStore _dataStoreForNSAttributedString]];
         [configuration setMediaTypesRequiringUserActionForPlayback:WKAudiovisualMediaTypeAll];
         [configuration _setAllowsJavaScriptMarkup:NO];
         [configuration _setAllowsMetaRefresh:NO];
         [configuration _setAttachmentElementEnabled:YES];
+#if HAVE(APP_SSO)
+        [configuration preferences]._extensibleSSOEnabled = NO;
```

Comment:
Why isn't that flag sufficient? Why do we need all the rest of this patch (the changes to the data store)?

Reply:
We have to create the WebsiteDataStore without AppSSO stuff, so it has to be done on line 142. However, I noticed that this flag existed, and was checked in some places, so I decided to also turn it off at the WKWebView configuration level in case something decided to spawn a web view or do something offscreen using its own data store.

I don't have evidence this is needed for this patch, but I think it is more correct to specify this (since it exists) than just trust that having the initial data store in the right state does the right thing at all times.

Reply:
> We have to create the WebsiteDataStore without AppSSO stuff

But why? Why isn't disabling SSO at WKWebView preferences level sufficient?

I looked at the code and the calls to `websiteDataStore().soAuthorizationCoordinator().tryAuthorize()` are completely bypasses when `[configuration preferences]._extensibleSSOEnabled` is set to `NO`. As a result, I don't understand why we need to make any changes to "disable" the soAuthorizationCoordinator() (since it should not be used in the first place).

Are you saying there is code using soAuthorizationCoordinator() without a WKWebView?

I may be missing something but this change still doesn't look right to me.